enhancements/etcd/automated-backups-no-config.md

```diff
@@ -0,0 +1,205 @@
+TBD
+#### Single-node Deployments or MicroShift
+TBD
+
+### Implementation Details/Notes/Constraints [optional]
+- Need to agree on a default schedule.
+- Need to agree on a default retention policy.
+
+- Several options exist for the default `PVCName`.
+  - Relying on `dynamic provisioning` is sufficient, however not an option for `SNO` or `BM` clusters.
+  - Utilising `local storage operator` is a proper solution, however installing a whole operator is too much overhead.
+  - The most viable solution to cover all OCP variants is to use `local volume`.
+    Please find below this solution's Pros & Cons.
+    - Pros :- 
+      - The PV will be bound to a specific master node.
+      - The Backup Pod will be scheduled always to this master node since it uses a PVC bound to this PV.
+      - Using an SSD is possible, since local volume allows mounting it into a Pod.
+      - Having the Backup Pod scheduling deterministic is vital, since the retention policy needs the recent backup in place to work.
+    - Cons :-
+      - If the master node where the PV is mounted became unhealthy/unavailable/unreachable etc.
+      - The backups are no longer accessible, also taking new backups is no longer possible.
+      - Ideally, the backups should be taken on a round-robin fashion on each master node to avoid overwhelming a specific etcd member more than others.
+      - Also spreading the backups among all etcd cluster members provide guarantees for disaster recovery in case of losing two members at the same time.
+      - However, using the `local volume` option over all the master nodes will be complicated and error prune.
+
+### Risks and Mitigations
+
+When the backups are configured to be saved to a `local` type PV, the backups are all saved to a single master node where the PV is provisioned on the local disk.
+
+In the event of the node becoming inaccessible or unscheduled, the recurring backups would not be scheduled. The periodic backup config would have to be recreated or updated with a different PVC that allows for a new PV to be provisioned on a node that is healthy.
+
+If we were to use `localVolume`, we need to find a solution for balancing the backups across the master nodes, some ideas could be
+
+- create a PV for each master node using `localVolume` and the backup controller from CEO should take care of balancing the backups across the `available` and `healthy` volumes.
+- the controller should keep the most recent backup on a healthy node available for restoration.
+- the controller should skip an unhealthy master node from taking a backup
+
+
+### Drawbacks
+
+## Design Details
+
+- Add default values to current `Etcdbackup Spec` using annotation.
+- Maintain the reconciliation logic with Cluster-etcd-operator.
+ 
+
+`Storage` solution used is based on two different designs, based on the OCP variant used 
+
+#### Cloud based OCP
+
+##### Pros
+- Utilising `Dynamic provisioning` is best option on cloud based OCP.
```

Comment:
how is the customer able to access a dynamically created PV to get their snapshots for restoration?

Reply:
Gr8 question, so I have defaulted the `RetentionPolicy` in my sample configs to `Retain`. 

This way the PV contents will never be erased automatically. 

Now to answer the restoration 

- If the cluster still running, the PV can be attached to any node, as long as it is in the same availability zone.
- If the cluster is completely down, the PV can be accessed by the CU, it will never be delete unless manually.

Reply:
it's not about the retention, it's about the access and mounting possibilities 

> If the cluster still running, the PV can be attached to any node, as long as it is in the same availability zone.

that's quite a bad constraint for a disaster recovery procedure, isn't it? :)

Reply:
:D :D  .. I know, well the best option imho is to push backups to **Remote storage** option, then even in case of the whole cluster is down, a new installation can be restored using the remote backup. 

However, in my discussion yesterday, the remote storage was not an option for BM. Also they recommended to keep it simple not too complicated 

If it were to me, I would create a side car which pushes the backups to remote storage, I think this option would work for any OCP variant. The master nodes dont have to be in the cloud, the side car could authenticate and push backups regardless of OCP underlying infrastructure, wdyt

Reply:
If dyanmically provisioned volumes are available in the cluster, it is 100% likely that, cluster has Remote Storage and hence backups are available even if cluster is down.

Reply:
I actually want to fall back to the `localVolume` approach, since having two separate approaches is harder to maintain. 

I believe the localVolume should be sufficient. However, if we could utilize STS across all the master nodes would be ideal in situation where we lose one or more master nodes.

Reply:
If I were to design this, I would definitely use two different backup strategies. It is far more reliable to use dynamically provisioned PVCs for backups. They can be snapshotted and can be accessible if node goes down. 

It seems to me that - any hostpath/localvolume based approach is basically inferior. If last backup was taken on leader and then leader node goes down, then so does our backup(other nodes may have slightly behind backups). So hostpath/localvolume requires fundamentally different recovery mechanism. 

IMO - if we take other components that use persistent storage in Openshift, if customers don't provide Storage then no persistent configuration is created. Prometheus or image-registry, they all require storage. And they don't do anything automatically.

So, what I am saying is - we should probably limit scope of this KEP only to environments where an StorageClass is available. If no StorageClass is configured then no automatic backups are taken. 

We should not take upon ourselves to decide a storage strategy for customer. cc @tjungblu

Reply:
So to summarize - I would simply not bother to configure backups via localvolume/hostpath in environments where no storage is available and solve that problem via documentation or ask customer to configure storage. I do not think, we should automatically configuring local-storage for these clusters.

Reply:
I actually agree that using dynamic provisioning and CSI snapshot is vital for the automated backup. 

I also agree to limit the `automated backup` for clusters with `dynamic provisioning` only. 

wdyt @tjungblu @hasbro17 @dusk125 @soltysh